### PR TITLE
kernel: move some symbols to generic config

### PR DESCRIPTION
**Pull request by abajk**
This commit add some enabled symbols to generic config.
LTO is only supported by clang compiler and therefore should be disabled in the generic config instead of duplicating this symbol in each target. CONFIG_LTO_NONE do this job.

The second group of symbols is enabled by the options available in the generic config and is therefore added here:
* CONFIG_AF_UNIX_OOB is selected by CONFIG_NET && CONFIG_UNIX,
* CONFIG_BINARY_PRINTF is selected by CONFIG_BPF_SYSCALL,
* CONFIG_NET_SOCK_MSG is selected by CONFIG_BPF_SYSCALL && CONFIG_NET.

The other symbols are disabled and should be in the generic config.

This commit also removes these symbols from subtargets.

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

### CODE REVIEW

File: target/linux/generic/config-5.15 [hunk 1]
```diff
@@ -3184,6 +3187,7 @@ CONFIG_LSM_MMAP_MIN_ADDR=65536
 # CONFIG_LTC2632 is not set
 # CONFIG_LTC2983 is not set
 # CONFIG_LTE_GDM724X is not set
+CONFIG_LTO_NONE=y
```
Comment (hauke) [inline target/linux/generic/config-5.15:3190]: Please remove this in the same commit from the target configurations:
```
target/linux/ipq806x/config-5.15:CONFIG_LTO_NONE=y
target/linux/mediatek/mt7622/config-5.15:CONFIG_LTO_NONE=y
target/linux/mediatek/mt7623/config-5.15:CONFIG_LTO_NONE=y
target/linux/ramips/mt7621/config-5.15:CONFIG_LTO_NONE=y
target/linux/sunxi/config-5.15:CONFIG_LTO_NONE=y
target/linux/x86/config-5.15:CONFIG_LTO_NONE=y
```

File: target/linux/generic/config-5.15 [hunk 2]
```diff
@@ -6868,7 +6878,9 @@ CONFIG_USB_GADGET_VBUS_DRAW=2
 # CONFIG_USB_MSI2500 is not set
 # CONFIG_USB_MSM_OTG is not set
 # CONFIG_USB_MTU3 is not set
+# CONFIG_USB_MUSB_GADGET is not set
```
Comment (hauke) [inline target/linux/generic/config-5.15:6881]: Please also remove these configurations for the target kernel configurations:
```
target/linux/mediatek/mt7623/config-5.10:# CONFIG_USB_MUSB_GADGET is not set
target/linux/mediatek/mt7623/config-5.15:# CONFIG_USB_MUSB_GADGET is not set
target/linux/omap/config-5.10:# CONFIG_USB_MUSB_GADGET is not set
target/linux/sunxi/cortexa53/config-5.10:# CONFIG_USB_MUSB_GADGET is not set
target/linux/sunxi/cortexa53/config-5.15:# CONFIG_USB_MUSB_GADGET is not set
target/linux/sunxi/cortexa7/config-5.10:# CONFIG_USB_MUSB_GADGET is not set
target/linux/sunxi/cortexa7/config-5.15:# CONFIG_USB_MUSB_GADGET is not set
```

I did not check the other configurations. Please remove the configurations from the individual target configurations when adding them to generic.

File: target/linux/generic/config-5.15 (hunk 1 above)
Comment (abajk) [reply target/linux/generic/config-5.15:3190]: Done

File: target/linux/generic/config-5.15 (hunk 2 above)
Comment (abajk) [reply target/linux/generic/config-5.15:6881]: donw

**Comment by hauke** [comment]
Thank you for your patch, I applied it to master in a98ded6c5c1891087fb43499829928b3bc17702f.

**Pull request closed without merging**